engine/types/type42/linking_key_test.go

```diff
@@ -0,0 +1,149 @@
+package type42
+
+import (
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/libsv/go-bk/bec"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCalculateHMAC(t *testing.T) {
+	tests := []struct {
+		name            string
+		pubSharedSecret []byte
+		reference       string
+		idx             int
+		expected        []byte
+		wantErr         bool
+	}{
+		{
+			name:            "valid input",
+			pubSharedSecret: []byte("shared_secret"),
+			reference:       "reference",
+			idx:             1,
+			expected:        computeExpectedHMAC([]byte("shared_secret"), "reference", 1),
+			wantErr:         false,
+		},
+		{
+			name:            "different index",
+			pubSharedSecret: []byte("shared_secret"),
+			reference:       "reference",
+			idx:             2,
+			expected:        computeExpectedHMAC([]byte("shared_secret"), "reference", 2),
+			wantErr:         false,
+		},
+		{
+			name:            "empty shared secret",
+			pubSharedSecret: []byte(""),
+			reference:       "reference",
+			idx:             1,
+			expected:        computeExpectedHMAC([]byte(""), "reference", 1),
+			wantErr:         false,
+		},
+		{
+			name:            "empty reference",
+			pubSharedSecret: []byte("shared_secret"),
+			reference:       "",
+			idx:             1,
+			expected:        computeExpectedHMAC([]byte("shared_secret"), "", 1),
+			wantErr:         false,
+		},
+		{
+			name:            "both empty",
+			pubSharedSecret: []byte(""),
+			reference:       "",
+			idx:             1,
+			expected:        computeExpectedHMAC([]byte(""), "", 1),
+			wantErr:         false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := calculateHMAC(tt.pubSharedSecret, fmt.Sprintf("%s-%d", tt.reference, tt.idx))
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expected, got)
+			}
+		})
+	}
+}
+
+func TestDeriveLinkedKey(t *testing.T) {
+	sourcePubKeyHex := "027c1404c3ecb034053e6dd90bc68f7933284559c7d0763367584195a8796d9b0e"
+	sourcePubKeyBytes, err := hex.DecodeString(sourcePubKeyHex)
+	assert.NoError(t, err)
```

Comment:
It should fail, even before any test if we have invalid pubkey

```suggestion
	require.NoError(t, err)
```

The same change applies to all assert.NoError below

Reply:
```suggestion
			assert "github.com/stretchr/testify/require"
```
done